iex_trading/IEX_stocks.ipynb

```diff
@@ -0,0 +1,451 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## IEX Stocks\n",
+    "\n",
+    "In the [previous notebook](./IEX_trading.ipynb) we saw how all the\n",
+    "trades on the IEX stock exchange could be interactively visualized over\n",
+    "the course of a whole day (Monday 21st of October 2019). Using\n",
+    "datashader, all the trades are rasterized interactively to reveal the\n",
+    "density of trades via a colormap.\n",
+    "\n",
+    "When viewing visualized a million trades at once for a whole day, it is\n",
+    "extremely difficult to identify individual trades using a global\n",
+    "view. In order to identify particular trades, it is necessary to zoom\n",
+    "into a time window small enough that individual trades can be\n",
+    "distinguished at which point the trade metadata can be inspected using\n",
+    "the Bokeh hover tool.\n",
+    "\n",
+    "What the global visualization helps reveal is the overall pattern of\n",
+    "trades. In this notebook we will focus on interactively revealing the\n",
+    "trading patterns for individual stocks by partitioning on a set of stock\n",
+    "symbols selected with a widget.\n",
+    "\n",
+    "## Loading the data\n",
+    "\n",
+    "First we will load the data as before, convert the integer timestamps\n",
+    "into the correctly offset datetimes before counting the total number of\n",
+    "events:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import datetime\n",
+    "import pandas as pd\n",
+    "df = pd.read_csv('./data/IEX_2019-10-21.csv')\n",
+    "df.timestamp = df.timestamp.astype('datetime64[ns]')\n",
+    "df.timestamp -= datetime.timedelta(hours=4)\n",
+    "print('Dataframe loaded containing %d events' % len(df))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next we will identify the top ten most traded stocks on this day and\n",
+    "compute how much of the trading volume they account for:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "top_symbol_counts = df.symbol.value_counts()[:10]\n",
+    "top_symbols = list(top_symbol_counts.index)\n",
+    "top_symbol_info = (', '.join(top_symbols), top_symbol_counts.sum() /len(df) * 100)\n",
```

Comment:
You've switched to summing the trade volumes so maybe this should also state the fraction of the overall volume?

Reply:
For that matter maybe select the top stocks by volume as well.

Reply:
Not a bad suggestion even though the number of trades is what tests datashader performance, not the volume. The top stocks by trade counts are what will push datashader the hardest here and any ten stocks will do as all I need to do is select between them. It might also be less cognitive overhead if I don't switch the names/symbols of the stocks inspected. @jbednar What do you think?

Reply:
Here you're distinguishing number of trades from number of stocks traded?  I guess I don't have much opinion; I'll leave it up to you as author of the example.